### PR TITLE
Add doc example for Module.eval_quoted with __ENV__ as first argument

### PR DESCRIPTION
**Pull request by nirev**
The usage with __ENV__ as the first argument to Module.eval_quoted was found in [Ecto's schema.ex](https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/schema.ex#L392) but that was not reflected in the docs.

I thought I might add an example for that. Is it ok?

### CODE REVIEW

**Comment by whatyouhide** [comment]
Maybe we should say that any `Macro.Env` struct is accepted, not just `__ENV__` (and options are extracted from that struct). Also, we should rename the first argument from `module` to `module_or_env` or something similar then :)

**Comment by nirev** [comment]
That's a good call, @whatyouhide. I'll add that! Also, what is the preferred way: several commits or just a single squashed one?

**Comment by whatyouhide** [comment]
@nirev we will squash the commits into one when merging the PR ([relatively recent GitHub feature](https://github.com/blog/2141-squash-your-commits)), so it's not much of a difference :)

**Comment by nirev** [comment]
Got it, thanks!

**Comment by nirev** [comment]
Should be better now! :)

File: lib/elixir/lib/module.ex [hunk 1]
```diff
@@ -406,8 +407,19 @@ defmodule Module do
...
+  which will have both the module and all options extracted from
+  the environment:
+
+      defmodule Foo do
+        contents = quote do: (def sum(a, b), do: a + b)
+        Module.eval_quoted __ENV__, contents
+      end
+
+      Foo.sum(1, 2) #=> 3
+
```
Comment (whatyouhide) [inline lib/elixir/lib/module.ex]: We could merge the two sections so as to not repeat most of the `Macro.Env`-related stuff. Wdyt?

File: lib/elixir/lib/module.ex [hunk 2]
```diff
@@ -396,8 +396,9 @@ defmodule Module do
 
       Foo.sum(1, 2) #=> 3
 
-  For convenience, you can pass `__ENV__/0` as an argument and
-  all options will be automatically extracted from the environment:
+  For convenience, you can pass any `Macro.Env` instance, such
```
Comment (whatyouhide) [inline lib/elixir/lib/module.ex]: "instance" is a very OO term 😛, what if we say "any `Macro.Env` struct"?

**Comment by nirev** [comment]
@whatyouhide I merged the two sections, but added a note about merging opts with `Macro.Env` right after you approved. Wdyt about the note?

**Comment by whatyouhide** [comment]
I approve even harder @nirev 💟

**Comment by nirev** [comment]
Awesome! Successful first elixir PR is successful! 
Thanks for taking the time to review such a small change! ❤️ 💚 💛 💙

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by whatyouhide** [comment]
Thank you @nirev, great work and great catch :) 👍

**Pull request merged**